src/object_type.h

```diff
@@ -19,10 +19,9 @@ static const ObjectType OBJECT_STATUE       =   2;    ///< Statue in towns
 static const ObjectType OBJECT_OWNED_LAND   =   3;    ///< Owned land 'flag'
 static const ObjectType OBJECT_HQ           =   4;    ///< HeadQuarter of a player
 
-static const ObjectType NUM_OBJECTS_PER_GRF = 255;    ///< Number of supported objects per NewGRF; limited to 255 to allow extending Action3 with an extended byte later on.
-
 static const ObjectType NEW_OBJECT_OFFSET   =   5;    ///< Offset for new objects
 static const ObjectType NUM_OBJECTS         = 64000;  ///< Number of supported objects overall
+static const ObjectType NUM_OBJECTS_PER_GRF = NUM_OBJECTS; ///< Number of supported objects per NewGRF
```

Comment:
does this constant need to exist anymore? as there's really no "per grf" limit, it's entirely covered by the global limit (if i'm understanding things correctly..)

Reply:
I left it in case we want to increase NUM_OBJECTS, as the per GRF limit would stay, although looking at it lots of that is limited to uint16 so it would be more involved anyway.